src/test/java/org/folio/rest/api/HoldingsStorageTest.java

```diff
@@ -813,6 +829,443 @@ public void cannotCreateHoldingWithoutPermanentLocation() throws Exception {
+
+    log.info("Finished cannotCreateAHoldingsWithHRIDFailure");
+  }
+
+  @Test
+  public void cannotChangeHRIDAfterCreation()
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+    log.info("Starting cannotChangeHRIDAfterCreation");
+
+    final UUID instanceId = UUID.randomUUID();
+    final UUID holdingsId = UUID.randomUUID();
+
+    instancesClient.create(smallAngryPlanet(instanceId));
+
+    setHoldingsSequence(1);
+
+    final JsonObject holdings = holdingsClient.create(new HoldingRequestBuilder()
+      .withId(holdingsId)
+      .forInstance(instanceId)
+      .withPermanentLocation(mainLibraryLocationId)
+      .withTags(new JsonObject().put("tagList", new JsonArray().add(TAG_VALUE)))).getJson();
+
+    assertThat(holdings.getString("hrid"), is("ho00000001"));
+
+    holdings.put("hrid", "ABC123");
+
+    final CompletableFuture<Response> updateCompleted = new CompletableFuture<>();
+
+    client.put(holdingsStorageUrl(String.format("/%s", holdingsId)), holdings, TENANT_ID,
+        text(updateCompleted));
+
+    final Response response = updateCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(response.getStatusCode(), is(HttpURLConnection.HTTP_BAD_REQUEST));
+    assertThat(response.getBody(),
+        is("The hrid field cannot be changed: new=ABC123, old=ho00000001"));
+
+    log.info("Finished cannotChangeHRIDAfterCreation");
+  }
+
+  @Test
+  public void cannotRemoveHRIDAfterCreation()
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+    log.info("Starting cannotRemoveHRIDAfterCreation");
+
+    final UUID instanceId = UUID.randomUUID();
+    final UUID holdingsId = UUID.randomUUID();
+
+    instancesClient.create(smallAngryPlanet(instanceId));
+
+    setHoldingsSequence(1);
+
+    final JsonObject holdings = holdingsClient.create(new HoldingRequestBuilder()
+      .withId(holdingsId)
+      .forInstance(instanceId)
+      .withPermanentLocation(mainLibraryLocationId)
+      .withTags(new JsonObject().put("tagList", new JsonArray().add(TAG_VALUE)))).getJson();
+
+    assertThat(holdings.getString("hrid"), is("ho00000001"));
+
+    holdings.remove("hrid");
+
+    final CompletableFuture<Response> updateCompleted = new CompletableFuture<>();
+
+    client.put(holdingsStorageUrl(String.format("/%s", holdingsId)), holdings, TENANT_ID,
+        text(updateCompleted));
+
+    final Response response = updateCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(response.getStatusCode(), is(HttpURLConnection.HTTP_BAD_REQUEST));
+    assertThat(response.getBody(),
+        is("The hrid field cannot be changed: new=null, old=ho00000001"));
+
+    log.info("Finished cannotRemoveHRIDAfterCreation");
+  }
+
+  @Test
+  public void canUsePutToCreateAHoldingsWhenHRIDNotSupplied()
```

Comment:
Similarly, can this be incorporated in the regular can put to create test?

Reply:
Same here, I like my unit tests to test just functionality I added and let existing tests ensure I didn't break any existing expectations. But, I can combine these tests as well.

Reply:
I can very much understand doing it in multiple steps as well. I quite often expand the test suite and then contract it in multiple commits (but then I tend to use smaller commits than most folks on FOLIO)